### PR TITLE
bpo-36210: correct logic in setup.py for optional extensions with regard to AIX

### PR DESCRIPTION
**Pull request by aixtools**
There are three optional modules that either will never exist (spwd and osaudiodev) or would build correctly if the OS provided library was preferred rather than an optionally installed, non-supported 3rd-party library (ncurses).

* This PR sets AIX to build _**_curses**_ against the IBM supported **/usr/lib/libcurses.a** and skip the optional _**_cursus_panel**_.
    * Python _curses module builds with no issues from AIX 5.3 through AIX 7.2 when using the officially supported libcurses.a
    * When a third-party packaging of ncurses is installed neither _**_curses**_ nor _**_cursus_panel**_ build successfully.

* setup.py no longer says _**_spwd**_ and _**_osaudiodev**_ are missing (something) when building for AIX

https://bugs.python.org/issue36210

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -869,7 +869,8 @@ def detect_readline_curses(self):
             curses_library = readline_termcap_library
         elif self.compiler.find_library_file(self.lib_dirs, 'ncursesw'):
             curses_library = 'ncursesw'
-        elif self.compiler.find_library_file(self.lib_dirs, 'ncurses'):
+        # Issue 36210: AIX and ncurses does not co-exist with IBM supplied curses
+        elif not HOST_PLATFORM.startswith("aix") and self.compiler.find_library_file(self.lib_dirs, 'ncurses'):
```
Comment (ZackerySpytz) [inline setup.py]: Please respect [PEP 8](https://www.python.org/dev/peps/pep-0008/) when adding new code. This line is too long.

Comment (aixtools) [reply setup.py]: nods. corrected.

**Comment by aixtools** [comment]
Question: what is the current status of spwd module in Python? Is this something I should look at seeing if I can emulate in AIX (i.e., not using getspent() etc, functions, but the AIX ones for accessing shadow information). 

I thought I heard that spwd will be removed - sometime.

If spwd is going to stay forever, then the missing announcement may be useful, but asis, it is just some noise.

**Comment by ncoghlan** [comment]
`spwd` is one of the libraries on the removal list in PEP 594: https://www.python.org/dev/peps/pep-0594/#spwd

File: setup.py [hunk 2]
```diff
@@ -989,6 +996,7 @@ def detect_readline_curses(self):
         elif curses_library == 'curses' and not MACOS:
                 # OSX has an old Berkeley curses, not good enough for
                 # the _curses module.
+            ncurses_enabled = False
```
Comment (ncoghlan) [inline setup.py]: The `ncurses_enabled` vs `curses_enabled` change is hard to follow, and hard to prove won't have unintended side effects on platforms other than AIX. I'd suggest replacing it with a simple `skip_curses_panel` flag that you set to true up on line 910, and then force to false in the AIX branch of the previous if-elif chain.

Then rather than checking for `ncurses_enabled` below, you would instead check for `not skip_curses_panel`.

**Review by ncoghlan — CHANGES_REQUESTED** [review]
This mostly seems fine to me, but I'd prefer to update the curses_panel logic to make it more obviously free of potential side effects on other platforms.

File: setup.py (hunk 2 above)
Comment (aixtools) [reply setup.py]: Except, I want to skip_curses_panel if AIX - so maybe it can be even simpler with a statement just before the curses_panel processing.

Comments please.

**Comment by aixtools** [comment]
I have made the requested changes; please review again

**Review by ncoghlan — APPROVED** [review]

**Comment by aixtools** [comment]
Added #16376 as backport.
